python/cudf/cudf/core/dataframe.py

```diff
@@ -2583,9 +2611,11 @@ def insert(self, loc, name, value, nan_as_null=None):
                         )
                 self._data = new_data
         elif isinstance(value, (pd.Series, Series)):
-            value = Series(value, nan_as_null=nan_as_null)._align_to_index(
-                self._index, how="right", sort=False
-            )
+            value = Series(value, nan_as_null=nan_as_null)
```

Comment:
There's probably a faster way to construct this if we know the input is a `cudf.Series`. I'm not sure how much we could save for a pandas Series by handling it manually, I don't think we typically do anything special for those. That may be worth exploring in a future PR (basically seeing if we can implement `Series.from_pandas` in a more efficient manner than just calling the constructor), but is out of scope for now.

Reply:
Yeah, I think it is out of scope for this PR. The reason being we have other places in the code-base which use the similar patterns we might be better off tackling that in a separate PR.